### ISSUE TITLE
Anny plan to support Pulumi

### ISSUE THREAD

**Issue by TechWatching** [radius-project/radius#6512]
## Overview of feature request

Allow to describe the applications using Pulumi.

## Acceptance criteria

The Radius application can be defined using Pulumi and not only bicep / Terraform.

**Comment by rynowak** [comment]
Hi @TechWatching - thanks for your question.

Part of our philosophy is that we're not opinionated about tools both for modeling the Application and for creating cloud resources in Recipes. It makes sense that we could support Pulumi (or any other tools) in the future if there's interest. 

I think what would be needed is for someone to build that support and maintain it going forward. We should keep this issue around and see what other feedback we get.

**Comment by rouke-broersma** [comment]
The way I understand Radius isn't this ask the wrong way around? Radius is built around ARM so as long as Pulumi implements the Radius RPs Pulumi would have support for Radius right? That seems the most efficient way to build out support to me. Or is that not the design direction Radius is looking for.

**Comment by AaronCrawfis** [comment]
> The way I understand Radius isn't this ask the wrong way around? Radius is built around ARM so as long as Pulumi implements the Radius RPs Pulumi would have support for Radius right? That seems the most efficient way to build out support to me. Or is that not the design direction Radius is looking for.

Yep this is the case, a Pulumi provider could be built for Radius using our ARM/UCP resource provider APIs for when you want to model a Radius application in Pulumi.

Separately, there's a Recipe Driver option for Pulumi, where Radius could itself deploy Pulumi code for Recipes, where a developer could write an application in Bicep/Terraform/Pulimi and under the hood infrastructure is deployed in Pulumi.
